### DISCUSSION TITLE
A question about Vector API in JDK 17

### DISCUSSION THREAD

**Discussion by attack204** [trinodb/trino#17828] [General]
Has there been any progress on the proposal to introduce JDK 17's Vector API in Trino since the issue was raised a year ago (https://github.com/trinodb/trino/issues/13186)? Is it still in the "todo" state, or has it been abandoned due to the Vector API not meeting expectations?

**Comment by martint** [comment]
There's no explicit goal to introduce the Vector API to Trino, but we'll do it as necessary where we can observe performance improvements. You can follow the larger efforts here: https://github.com/trinodb/trino/issues/14237

The biggest challenge at this point in time is that the API is still in "preview" mode and subject to change from version to version of the JDK, so it needs to be done carefully to avoid breaking backward compatibility if someone decides to run a different build of the JDK.
